presto-main/src/main/java/com/facebook/presto/sql/planner/DependencyExtractor.java

```diff
@@ -43,6 +44,14 @@ private DependencyExtractor() {}
         return uniqueSymbols.build();
     }
 
+    public static Set<Symbol> extractUniqueNonRecursive(PlanNode node)
```

Comment:
The real issue is that extractUnique is including "local" symbols. The correct behavior would be to do it recursively, but only include the output symbols of `node` and any symbol that comes from a correlation for any child node.